### PR TITLE
driver(adapters): More test fixes focused on neon/pg

### PR DESCRIPTION
**Pull request by miguelff**
This fixes a few unsupported types

### CODE REVIEW

File: query-engine/driver-adapters/src/proxy.rs [hunk 1]
```diff
@@ -317,6 +322,14 @@ fn js_value_to_quaint(
                 column_name, mismatch
             ),
         },
+        ColumnType::Uuid => match json_value {
+            serde_json::Value::String(s) => {
+                let s_as_uuid = uuid::Uuid::parse_str(&s).expect("Expected a UUID string");
+                QuaintValue::uuid(s_as_uuid)
+            }
```
Comment (aqrln) [inline query-engine/driver-adapters/src/proxy.rs:329]:
```suggestion
            serde_json::Value::String(s) => uuid::Uuid::parse_str(&s)
                .map(QuaintValue::uuid)
                .map_err(conversion_error!("Expected a UUID string"))
```

File: query-engine/driver-adapters/src/proxy.rs [hunk 2]
```diff
@@ -317,6 +322,14 @@ fn js_value_to_quaint(
...
+            serde_json::Value::Null => QuaintValue::Bytes(None),
```
Comment (aqrln) [inline query-engine/driver-adapters/src/proxy.rs:330]:
```suggestion
            serde_json::Value::Null => Ok(QuaintValue::Bytes(None)),
```

File: query-engine/driver-adapters/src/proxy.rs [hunk 3]
```diff
@@ -317,6 +322,14 @@ fn js_value_to_quaint(
...
+            mismatch => panic!("Expected a UUID string in column {}, found {}", column_name, mismatch),
```
Comment (aqrln) [inline query-engine/driver-adapters/src/proxy.rs:331]:
```suggestion
            mismatch => Err(conversion_error!("Expected a UUID string in column {}, found {}", column_name, mismatch)),
```

**Review by aqrln — COMMENTED** [review]
heads up: https://github.com/prisma/prisma-engines/pull/4300 now merged, github doesn't report conflicts but the code will need to be updated

**Comment by miguelff** [comment]
@aqrln could you please review, adapt this to latest changes and merge?

**Review by aqrln — APPROVED** [review]

**Pull request merged**